src/metabase/models/secret.clj

```diff
@@ -243,8 +242,8 @@
         latest-version (when existing-id (latest-for-id existing-id))]
     (if latest-version
       (if (= (select-keys latest-version bump-version-keys) [kind src value])
-        (db/update-where! Secret {:id existing-id :version (:version latest-version)}
-                                 :name nm)
+        (t2/update! Secret {:id existing-id :version (:version latest-version)}
+                    {:name nm})
```

Comment:
To preserve existing behaviour, you should call `pos?` on the return value.

`upsert-secret-value!` only used once, [here](https://github.com/metabase/metabase/blob/e50cd5dfb4b1dcbbc60ace581c5de1f306307f6b/src/metabase/models/database.clj#L132). The destructured `secret*` value could be an integer instead of a boolean now. It gets passed to `expand-inferred-secret-values` as a `secret-or-id` argument, and because it's an integer [this predicate ](https://github.com/metabase/metabase/blob/e50cd5dfb4b1dcbbc60ace581c5de1f306307f6b/src/metabase/models/secret.clj#L293) will now return true, possibly leading to the wrong secret being used for the database. I couldn't find a test for this.

Reply:
good catch. I'll wrap it with a `pos?`